File: app/src/main/java/com/oriente/aptsample/MainActivity.java
package com.oriente.aptsample;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.example.user.ThirdActivity$Route;
import com.oriente.anno.BindView;
import com.oriente.anno.Route;
import com.oriente.anno.RouteInfo;

import java.util.List;

@Route(name = "MainActivity")
public class MainActivity extends AppCompatActivity {


    @BindView(R.id.text_dashboard)
    TextView text_dashboard;

    @BindView(R.id.text2)
    TextView text2;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        setTitle("MainActivity");
        findViewById(R.id.text_dashboard).setOnClickListener(v -> {
            Class target = ThirdActivity$Route.getTarget("ThirdActivity");
            startActivity(new Intent(this, target));

        });


//        ButterKnife.bind(this);
//
//        StringBuffer sb = new StringBuffer();
//        List<RouteInfo> sRouterTable = RouterTable.sRouterTable;
//        for (RouteInfo routeInfo : sRouterTable) {
//            sb.append(routeInfo.name).append("\n");
//        }
//        text2.setText(sb.toString());
    }
}

File: app/src/main/java/com/oriente/aptsample/MainActivity.kt
package com.oriente.aptsample

import android.os.Bundle
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.lifecycleScope
import com.oriente.aptsample.databinding.ActivityMainBinding
import com.oriente.aptsample.viewmodel.UserModel
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch

/**
 * LiveData + ViewModel + DataBinding + CoroutineScope
 */

class MainActivity : AppCompatActivity() {

    private val userModel by viewModels<UserModel>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding: ActivityMainBinding =
            DataBindingUtil.setContentView(this, R.layout.activity_main)

        binding.userModel = userModel
        binding.lifecycleOwner = this

        lifecycleScope.launch {
            repeat(10) {
                delay(2000)
                userModel.age.postValue(userModel.age.value?.plus(100) ?: 10)
                userModel.name.postValue(userModel.name.value + "alex")
            }
        }
    }
}